packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts

```diff
@@ -0,0 +1,148 @@
+import { createResolvedObject, getOriginalObject } from './resolvedObject';
+
+describe('createResolvedObject', () => {
+  it('resolves a reference', () => {
+    const resolvedObject = createResolvedObject({
+      paramaterA: {
+        parameterB: {
+          $ref: '#/bundled/parameterB',
+        },
+      },
+      bundled: {
+        parameterB: 'parameterB value',
+      },
+    });
+
+    expect((resolvedObject as any).paramaterA.parameterB).toBe('parameterB value');
+  });
+
+  it('resolves deeply nested reference', () => {
+    const resolvedObject = createResolvedObject({
+      paramaterA: {
+        parameterB: {
+          $ref: '#/bundled/parameterB',
+        },
+      },
+      bundled: {
+        parameterB: {
+          parameterC: {
+            $ref: '#/bundled/parameterC',
+          },
+        },
+        parameterC: 'parameterC value',
+      },
+    });
+
+    expect((resolvedObject as any).paramaterA.parameterB.parameterC).toBe('parameterC value');
```

Comment:
Can we try a recursive one? 😇 

I mean, adding a new test of course. Something that references itself. then 
```ts
expect(resolvedObject.someProp.otherProp.someProp.otherProp...).toEqual(resolvedObject)
```